docs/proposals/20201020-management-cluster-operator.md

```diff
@@ -0,0 +1,1038 @@
+  * [Glossary](#glossary)
+  * [Summary](#summary)
+  * [Motivation](#motivation)
+     * [Goals](#goals)
+     * [Non-Goals/Future Work](#non-goalsfuture-work)
+  * [Proposal](#proposal)
+     * [User Stories](#user-stories)
+     * [Implementation Details/Notes/Constraints](#implementation-detailsnotesconstraints)
+        * [Existing API Types Changes](#existing-api-types-changes)
+        * [New API Types](#new-api-types)
+        * [Example API Usage](#example-api-usage)
+        * [Operator Behaviors](#operator-behaviors)
+           * [Installing a provider](#installing-a-provider)
+           * [Upgrading a provider](#upgrading-a-provider)
+           * [Upgrades providers without changing contract](#upgrades-providers-without-changing-contract)
+           * [Upgrades providers and changing contract](#upgrades-providers-and-changing-contract)
+           * [Changing a provider](#changing-a-provider)
+           * [Deleting a provider](#deleting-a-provider)
+        * [Upgrade from v1alpha3 management cluster to v1alpha4 cluster](#upgrade-from-v1alpha3-management-cluster-to-v1alpha4-cluster)
+        * [Operator Lifecycle Management](#operator-lifecycle-management)
+           * [Operator Installation](#operator-installation)
+           * [Operator Upgrade](#operator-upgrade)
+           * [Operator Delete](#operator-delete)
+        * [Air gapped environment](#air-gapped-environment)
+     * [Risks and Mitigation](#risks-and-mitigation)
+        * [Error Handling &amp; Logging](#error-handling--logging)
+        * [Extensibility Options](#extensibility-options)
+        * [Upgrade from v1alpha3 management cluster to v1alpha4/operator cluster](#upgrade-from-v1alpha3-management-cluster-to-v1alpha4operator-cluster)
+  * [Additional Details](#additional-details)
+     * [Test Plan](#test-plan)
+     * [Version Skew Strategy](#version-skew-strategy)
+  * [Implementation History](#implementation-history)
+  * [Controller Runtime Types](#controller-runtime-types)
+
+## Glossary
+
+The lexicon used in this document is described in more detail
+[here](https://github.com/kubernetes-sigs/cluster-api/blob/master/docs/book/src/reference/glossary.md).
+Any discrepancies should be rectified in the main Cluster API glossary.
+
+## Summary
+
+The clusterctl CLI currently handles the lifecycle of Cluster API
+providers installed in a management cluster. It provides a great Day 1
+experience in getting CAPI up and running. However, clusterctl’s imperative
+design makes it difficult for cluster admins to stand up and manage CAPI
+management clusters in their own preferred way.
+
+This proposal provides a solution that leverages a declarative API and an
+operator to empower admins to handle the lifecycle of providers within the
+management cluster
+
+## Motivation
+
+In its current form clusterctl is designed to provide a super simple user
+experience for day 1 operations of a Cluster API management cluster with a
+single infrastructure provider.
```

Comment:
I've decided to remove the part `with a single infrastructure provider.` since this is false. We can do `clusterctl init -i aws,azure` to install CAPA and CAPZ for example.